tests/recipes/recipe_lib_test.py

```diff
@@ -0,0 +1,151 @@
+from unittest import mock
+from tests.recipes.recipe_ctx import RecipeCtx
+
+
+class BaseTestForMakeRecipe(RecipeCtx):
+    """
+    An unittest for testing any recipe using the standard build commands
+    (`configure/make`).
+
+    .. note:: Note that Some cmake recipe may need some more specific testing
+        ...but this should cover the basics.
+    """
+
+    recipe_name = None
+    recipes = ["python3", "kivy"]
+    recipe_build_order = ["hostpython3", "python3", "sdl2", "kivy"]
+    expected_compiler = (
+        "{android_ndk}/toolchains/llvm/prebuilt/linux-x86_64/bin/clang"
+    )
+
+    sh_command_calls = ["./configure"]
+    """The expected commands that the recipe runs via `sh.command`."""
+
+    extra_env_flags = {}
+    """
+    This must be a dictionary containing pairs of key (env var) and value.
+    """
+
+    def __new__(cls, *args):
+        obj = super(BaseTestForMakeRecipe, cls).__new__(cls)
+        if obj.recipe_name is not None:
+            print(f"We are testing recipe: {obj.recipe_name}")
+            obj.recipes.append(obj.recipe_name)
```

Comment:
It always confuse me how Python can deal with class attributes vs instance attributes.
For instance initially `recipes` is a class attribute, right? Then we access it from the instance to alter it. Would this modify also the class attribute or just for this instance? :smile:

Reply:
Well, [class attributes are supposed to be shared for all instances](https://www.geeksforgeeks.org/class-instance-attributes-python/)...so  I guess that the answer is yes, we are also modifying class attributes (the thing is that recipes will grow up as long as we perform the tests...but shouldn't affect the test itself at all)

Reply:
Exactly, in this case why not making it explicit and access it through `cls`? That would avoid confusion. Do you see what I mean? This is rather nitpicking, but at the same time I personally feel it's important to get the habit about thinking/clarifying theses when possible since it can cause nasty side effects sometimes

Reply:
Yes you are right, I'm preparing another PR to address all this we recently talk that cannot be done here. Later I will push it.

**Thanks again** @AndreMiras, your reviews always makes the code better :+1: